webapp/app/js/compareperf.js

```diff
@@ -326,19 +370,29 @@ perf.controller('CompareSubtestResultsCtrl', [
                   PhSeries.getSubtestSummaries($scope.newProject,
                                 timeRange,
                                 optionCollectionMap,
-                                $scope.originalSignature).then(
+                                $scope.newSignature).then(
                     function (newSeriesData) {
+                      $scope.platformList = _.union($scope.platformList,
+                                                    newSeriesData.platformList).sort();
+                      $scope.testList = _.union($scope.testList,
+                                                    newSeriesData.testList).sort();
+
                       return PhCompare.getResultsMap($scope.newProject,
                                            newSeriesData.seriesList,
                                            timeRange,
                                            [$scope.newResultSetID]);
                   }).then(function(newSeriesMaps) {
                     var newSeriesMap = newSeriesMaps[$scope.newResultSetID];
-                    Object.keys(newSeriesMap).forEach(function(series) {
-                      if (!_.contains($scope.pageList, newSeriesMap[series].name)) {
-                        $scope.pageList.push(newSeriesMap[series].name);
-                      }
-                    });
+                    if (newSeriesMap) {
```

Comment:
updated the PR to address feedback!

On Wed, Apr 29, 2015 at 12:07 PM, William Lachance <notifications@github.com
> wrote:

> In webapp/app/js/compareperf.js
> <https://github.com/mozilla/treeherder-ui/pull/495#discussion_r29351335>:
>
> >                        return PhCompare.getResultsMap($scope.newProject,
> >                                             newSeriesData.seriesList,
> >                                             timeRange,
> >                                             [$scope.newResultSetID]);
> >                    }).then(function(newSeriesMaps) {
> >                      var newSeriesMap = newSeriesMaps[$scope.newResultSetID];
> > -                    Object.keys(newSeriesMap).forEach(function(series) {
> > -                      if (!_.contains($scope.pageList, newSeriesMap[series].name)) {
> > -                        $scope.pageList.push(newSeriesMap[series].name);
> > -                      }
> > -                    });
> > +                    if (newSeriesMap) {
>
> Under what conditions would newSeriesMap not be defined? It kind of looks
> like we might be papering over something that shouldn't happen?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mozilla/treeherder-ui/pull/495/files#r29351335>.
>